qcodes/tests/dataset/temporary_databases.py

```diff
@@ -15,71 +14,69 @@
 
 
 @pytest.fixture(scope="function")
-def empty_temp_db():
+def empty_temp_db(tmp_path):
     global n_experiments
     n_experiments = 0
     # create a temp database for testing
-    with tempfile.TemporaryDirectory() as tmpdirname:
-        try:
-            qc.config["core"]["db_location"] = \
-                os.path.join(tmpdirname, 'temp.db')
-            if os.environ.get('QCODES_SQL_DEBUG'):
-                qc.config["core"]["db_debug"] = True
-            else:
-                qc.config["core"]["db_debug"] = False
-            initialise_database()
-            yield
-        finally:
-            # there is a very real chance that the tests will leave open
-            # connections to the database. These will have gone out of scope at
-            # this stage but a gc collection may not have run. The gc
-            # collection ensures that all connections belonging to now out of
-            # scope objects will be closed
-            gc.collect()
+    try:
+        qc.config["core"]["db_location"] = \
+            str(tmp_path / 'temp.db')
+        if os.environ.get('QCODES_SQL_DEBUG'):
+            qc.config["core"]["db_debug"] = True
+        else:
+            qc.config["core"]["db_debug"] = False
+        initialise_database()
+        yield
+    finally:
+        # there is a very real chance that the tests will leave open
+        # connections to the database. These will have gone out of scope at
+        # this stage but a gc collection may not have run. The gc
+        # collection ensures that all connections belonging to now out of
+        # scope objects will be closed
+        gc.collect()
```

Comment:
We can perhaps drop these now?

Reply:
we could try. but as per my understanding this really ensures that connections that are "out of scope" are closed, which allows us to detect those that may still be open (due to some stupid bug or bad design)

Reply:
Yes I will leave this for now